name-clusters/script.whizzml

```diff
@@ -0,0 +1,121 @@
+;; model-importances
+;;
+;; Returns importances from the model
+;;
+;; Inputs:
+;;   model-id: (map) Corresponding model ID
+;;
+;; Output: (list) String representing the field = value name
+(define (model-importances model-id)
+  (let (model (fetch model-id)
+        importances-ids (model ["model" "importance"]))
+    (map (lambda (n) (head n))
+         importances-ids)))
+
+
+;; generate-models
+;;
+;; Returns a list of the model IDs for the clusters when finished
+;;
+;; Inputs:
+;;   cluster-id: (string) Cluster ID
+;;   centroid-ids: (list) List of centroid IDs
+;;
+;; Output: (list) String representing the field = value name
+(define (generate-models cluster-id centroid-ids)
+  (let (cluster (fetch cluster-id)
+        models (cluster "cluster_models" {})
+        model-ids (for (centroid-id centroid-ids)
+                    (if (models centroid-id false)
+                        (str "model/" (models centroid-id))
+                        (create "model" {"cluster" cluster-id
+                                         "centroid" centroid-id}))))
+    (wait* model-ids)))
+
+
+;; cluster-name
+;;
+;; Creates cluster name  using the most important field names of its
+;; correspondent model and the centroid values
+;;
+;; Inputs:
+;;   clusters: (map) Structure of changes to set the names
+;;   centroid: (map) Centroid info
+;;   fields: (map) Cluster fields
+;;   model-id: (map) Model ID
+;;   number-of-terms: (number) Number of terms to use in cluster name
+;;   separator: (string) Separator to join terms to form cluster names
+;;
+;; Outputs: (string) Cluster name
+;;
+(define (cluster-name clusters
+                      centroid
+                      fields
+                      model-id
+                      number-of-terms
+                      separator)
+  (let (importances-ids (model-importances model-id)
+        name-strs (map (lambda (x) (str (fields [x "name"] "")
+                                        " = "
+                                        (centroid ["center" x])))
+                       importances-ids)
+        n (- number-of-terms 1)
+    ;; List of already named clusters to avoid repeating names
+        seen (map (lambda (c) (c "name")) (values clusters)))
```

Comment:
you don't update the list of `seen` names in the loop below, as you add new ones: is that right?

Reply:
I am not sure if I am getting the point, but this loop is inside a function which is executed for each cluster name. On each function execution, seen contains the updated data of previous name clusters.